beps/0009-plugin-metadata/README.md

```diff
@@ -0,0 +1,173 @@
+
+Each plugin package must define a `backstage.pluginId` field, which is the same identifier as is used in the implementation of the plugin. This field is inferred from the package name by the `backstage-cli repo fix` command if it is not present. It should only be defined for plugin package, for example `@backstage/errors` should not define a plugin ID. The `backstage.pluginId` field is required when publishing a package with a plugin or module role, or a library role with "plugin" in its name.
+
+The package relationships are defined in the `backstage.pluginPackages` field. The value of the field is an object where each key is the role of the package as defined by the `backstage.role` field, and the value is the package name. For example:
+
+```json
+{
+  "name": "@backstage/plugin-catalog",
+  "backstage": {
+    "role": "frontend-plugin",
+    "pluginId": "catalog",
+    "pluginPackages": {
+      "frontend-plugin": "@backstage/plugin-catalog",
+      "backend-plugin": "@backstage/plugin-catalog-backend",
+      "web-library": "@backstage/plugin-catalog-react",
+      "node-library": "@backstage/plugin-catalog-node",
+      "common-library": "@backstage/plugin-catalog-common"
+    }
+  }
+}
+```
+
+The `backstage.pluginPackages` field is generated and updated by the `backstage-cli repo fix` command based on the packages that are present in the workspace and their `backstage.pluginId` and `backstage.role` fields. There can only be a single package of each role with a given plugin ID. The `backstage.pluginPackages` field is required when publishing a package with a `backstage.pluginId` field that is not using a module role.
+
+Module packages define their target plugin both via the `backstage.pluginId` field, as well as via `backstage.pluginPackage`. For example:
+
+```json
+{
+  "name": "@backstage/plugin-catalog-backend-module-github",
+  "backstage": {
+    "role": "backend-plugin-module",
+    "pluginId": "catalog",
+    "pluginPackage": "@backstage/plugin-catalog-backend"
+  }
+}
+```
+
+The `backstage.pluginPackage` field is required when publishing a package with a module role.
+
+## Design Details
+
+_-_
+
+### References & Prior Art
+
+The following resources have been used to inform this proposal:
+
+- [package.json documentation](https://docs.npmjs.com/cli/v10/configuring-npm/package-json)
+- [VSCode Extensions](https://code.visualstudio.com/api/references/extension-manifest)
+
+## Release Plan
+
+TBD
+
+<!--
+This section should describe the rollout process for any new features. It must take our version policies into account and plan for a phased rollout if this change affects any existing stable APIs.
+
+If there is any particular feedback to be gathered during the rollout, this should be described here as well.
+-->
+
+## Dependencies
+
+None
+
+<!--
+List any dependencies that this work has on other BEPs or features.
+-->
+
+## Alternatives
+
+TBD
```

Comment:
An alternative is to establish a wholly separate file from `package.json`, as related-but-not-dependent packages are not something npm has a good solution for. There could be a `plugin.json` file that lives outside of a single package (in the monorepo root?) and references packages from the outside.

This alternative has a glaring flaw in that it wouldn't be part of package publishing — unless you copy the file into each package as it's published, which is then very close to the same solution.

An advantage to this approach is that a separate file might feel less like abuse of `package.json`, especially if more fields are added over time. We've considered before that such a plugin metadata file might be useful for plugin listings such as the [io plugin list](https://backstage.io/plugins/). For that purpose, you might want to store a plugin name, description, and publisher, for example.